packages/astro/src/core/build/generate.ts

```diff
@@ -488,7 +491,12 @@ function getUrlForPath(
 	return url;
 }
 
-async function generatePath(pathname: string, gopts: GeneratePathOptions, pipeline: BuildPipeline) {
+async function generatePath(
+	pathname: string,
+	gopts: GeneratePathOptions,
+	pipeline: BuildPipeline,
```

Comment:
The `pipeline` has already the information you need, you don't need to pass `siteURL` all around.

Just call `pipeline.getConfig()`